lib/internal/process/warning.js

```diff
@@ -7,18 +7,21 @@ const {
 
 const assert = require('internal/assert');
 const { ERR_INVALID_ARG_TYPE } = require('internal/errors').codes;
+const path = require('path');
 
 // Lazily loaded
 let fs;
 let fd;
 let warningFile;
+let options;
 
 function lazyOption() {
   // This will load `warningFile` only once. If the flag is not set,
   // `warningFile` will be set to an empty string.
   if (warningFile === undefined) {
-    warningFile = require('internal/options')
-                  .getOptionValue('--redirect-warnings');
+    options = require('internal/options');
+    warningFile = path.resolve(options.getOptionValue('--diagnostic-dir'),
+                               options.getOptionValue('--redirect-warnings'));
```

Comment:
I think this is causing (at least some) of the CI test failures. If no options are set `getOptionValue('--redirect-warnings')` returns `''` (i.e. the empty string). 

In the before case `warningFile` is therefore set to `''` and the later `if (warningFile)` test fails. 

`path.resolve('', '')` resolves to the current working directory, so this PR has changed `warningFile` to a non-empty string, resulting in the warning being redirected instead of sent to `console.error`. I'm not entirely sure why non-Windows platforms are not failing.

Reply:
> 
> `path.resolve('', '')` resolves to the current working directory, so this PR has changed `warningFile` to a non-empty string, resulting in the warning being redirected instead of sent to `console.error`. I'm not entirely sure why non-Windows platforms are not failing.

Ah that's because the `fs.openSync(warningFile, 'a')` call is in a try...catch block where the catch redirects the warning back to `console.error`. So for some reason Windows isn't throwing (i.e. the open succeeded?).